### ISSUE TITLE
[VarArgs] unexpected behaviour of returnsArgAt(int)

### ISSUE THREAD

**Issue by ChristianSchwarz** [mockito/mockito#820]
I stumble about a test that failed after refactoring an array parameter to an varargs.

The following test succeed:
```java
interface Foo{
        int[] getOdd(int[] input);
}

@Test
public void test() throws Exception {
    when(foo.getOdd(any())).then(returnsArgAt(0));
        
    int[] odd = mock.getOdd(new int[]{1,3,5});
    assertThat(odd).contains(1,3,5);
}
```
Now lets change the parameter _input_ to an varargs.
```java
interface Foo{
        int[] getOdd(int... input);
}
```
The test fails now with:

```
org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 
The argument of type 'int' cannot be returned because the following 
method should return the type 'int[]'
 -> foo.getOdd()
```

This is not expected cause the change is source and binary(?) compatible, an varargs-parameter is still an array. ReturnsArgumentAt interprets the vararg array items as real parameters. In other words it tries to return an element type where the array type is required.

Can the behaviour of interpreting vararg elements like parameters be changed so an vararg is treated as an array? E.g. in Mockito 3? There are several other cases were the current vararg handling leads to unexpected behaviour.

**Comment by TimvdLippe** [comment]
I think your second example has a typo, since there are no var args in the example

**Comment by ChristianSchwarz** [comment]
thanks fixed

**Comment by bric3** [comment]
Good catch!
By the way, the mock reference is both foo and mock

**Comment by bric3** [comment]
While it is source and binary compatible, vararg method are different than plain old array method, reflection allows to distinguish that. The issue is in `ReturnsArgumentAt.returnedTypeOnSignature` where there's a special handling for varargs.

**Comment by bric3** [comment]
I'm working on the fix (currently adding some unit tests)

**Comment by bric3** [comment]
This should be fixed by #821
